native/src/modules/categories/components/CategoryListContent.js

```diff
@@ -0,0 +1,181 @@
+import styled from 'styled-components/native'
+import { type StyledComponent } from 'styled-components'
+import type { NavigationStackProp } from 'react-navigation-stack'
+import type { NavigateToInternalLinkParamsType } from '../../app/createNavigateToInternalLink'
+import type { ThemeType } from '../../../../build-configs/ThemeType'
+import { RTL_LANGUAGES } from '../../i18n/constants'
+import onLinkPress from '../../common/onLinkPress'
+
+const VerticalPadding: StyledComponent<{}, {}, *> = styled.View`
+  padding: 0 20px;
+`
+
+// type inferred from 'react-native/Libraries/Utilities/Dimensions'
+type DimensionsEventType = {
+  window?: DisplayMetrics,
+  screen?: DisplayMetrics,
+  ...
+}
+
+type ContentPropsType = {|
+  content: string,
+  navigation: NavigationStackProp<*>,
+  resourceCacheUrl: string,
+  navigateToInternalLink?: NavigateToInternalLinkParamsType => void,
+  cacheDictionary: { [string]: string },
+  language: string,
+  theme: ThemeType
+|}
+
+class CategoryListContent extends React.Component<ContentPropsType, {| width: number |}> {
+  constructor () {
+    super()
+    this.state = {
+      width: Dimensions.get('window').width
+    }
+  }
+
+  onChange = (dimensionsEvent: DimensionsEventType) => {
+    const defaultWidth = 200
+    this.setState({ width: (dimensionsEvent.window?.width || dimensionsEvent.screen?.width || defaultWidth) })
+  }
+
+  componentDidMount () {
+    Dimensions.addEventListener('change', this.onChange)
+  }
+
+  componentWillUnmount () {
+    Dimensions.removeEventListener('change', this.onChange)
+  }
+
+  onLinkPress = (evt: GestureResponderEvent, url: string) => {
+    const { language, navigation, navigateToInternalLink } = this.props
+    onLinkPress(url, navigation, language, navigateToInternalLink)
+  }
+
+  alterResources = (node: HTMLNode) => {
+    const { cacheDictionary } = this.props
+    if (node.attribs) {
+      if (node.attribs.href) {
+        console.debug(`Found href: ${decodeURI(node.attribs.href)}`)
+        const newResource = cacheDictionary[decodeURI(node.attribs.href)]
+        if (newResource) {
+          console.debug(`Replaced ${node.attribs.href} with ${newResource}`)
+          node.attribs = {
+            ...(node.attribs || {}),
+            href: newResource
+          }
+          return node
+        }
+      } else if (node.attribs.src) {
+        console.debug(`Found src: ${decodeURI(node.attribs.src)}`)
+        const newResource = cacheDictionary[decodeURI(node.attribs.src)]
+        if (newResource) {
+          console.debug(`Replaced ${node.attribs.src} with ${newResource}`)
+          node.attribs = {
+            ...(node.attribs || {}),
+            src: newResource
+          }
+          return node
+        }
+      }
+    }
+  }
+
+  // see https://github.com/archriss/react-native-render-html/issues/286
+  renderBulletLists: RendererFunction = (htmlAttribs, children, convertedCSSStyles, passProps) => {
```

Comment:
This function is too complicated imo. There are to many conditions. This should be refactored imo.

Reply:
As this is just a workaround for a not implemented functionality of the library (RTL lists) I think this is not worth the effort to be honest.